### PR TITLE
chore: update manifests to 0.13.1

### PR DESCRIPTION
**Pull request by bwplotka**
Cherry pick of https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1117/files#diff-812b265e1da8d8a3c64370d7d4b87bd6d109bda0d44b066ee86b8329d4004f43 and distroless bump.

### CODE REVIEW

File: charts/values.global.yaml [hunk 1]
```diff
@@ -16,29 +16,31 @@ commonLabels: false
...
   system: gmp-system
-version: 0.13.0
+version: 0.13.1
 images:
+  # NOTE: All tags have to be quoted otherwise they will be treated as a number.
   bash:
     image: gke.gcr.io/gke-distroless/bash
-    tag: "gke_distroless_20240607.00_p0" # NOTE: Has to be quoted otherwise it will be treated as a number.
+    tag: "gke_distroless_20240907.00_p0"
   alertmanager:
```
Comment (damemi) [inline charts/values.global.yaml:25]: is it possible to update the alertmanager image at this stage of the release or no?

**Comment by pintohutch** [comment]
suggestion: `git cherry-pick` https://github.com/GoogleCloudPlatform/prometheus-engine/commit/3d89ecba2a3841620b4a8021b688b2ed9788d531

File: charts/values.global.yaml (hunk 1 above)
Comment (bwplotka) [reply charts/values.global.yaml:25]: Not yet, we will do once 0.14.0 is out with manually bumped AM new image.

Comment (bwplotka) [reply charts/values.global.yaml:25]: However we can follow what @pintohutch suggested and did in 0.13 branch - change things on release/0.14 to what we expect to release once prod images are there, then cherry pick to main once done. Will try that.

**Comment by bwplotka** [comment]
Nice I like your flow started in https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1117/files#diff-812b265e1da8d8a3c64370d7d4b87bd6d109bda0d44b066ee86b8329d4004f43

Cherry-pick + update of distroless here. PTAL @pintohutch @damemi

**Comment by bwplotka** [comment]
I am going to merge this. I think we have to simply claim that manifests from main will be not always buildable due to tag lag https://github.com/GoogleCloudPlatform/prometheus-engine/issues/1013#issuecomment-2363252303

0.13.1 is good to have here, but due to perms main manifests won't work anyway, proceeding cutting 0.14 release branch to get 0.14 prod images we can cherry pick later to main.

**Pull request merged**